File: lc367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        ans=False
        for i in range(1,num+1):
            if i*i==num:
                ans=True
                break
            if i*i>num:
                break
        return ans
print(Solution.isPerfectSquare(14))

File: lc367.java
public class lc367 {
    //    367. Valid Perfect Square
    public static void main(String[] args) {
        int num = 1;
        boolean ans = false;
        if (num == 1) {
            ans = true;
        } else {
            for (int i = 1; i <= num / 2; i++) {
                if (i * i == num) {
                    ans = true;
                    break;
                }
                if (i * i > num) {
                    break;
                }
            }
        }
        System.out.println(ans);
    }
}
